src/internal/sio_client_impl.cpp

```diff
@@ -577,10 +577,9 @@ namespace sio
 #if SIO_TLS
     client_impl::context_ptr client_impl::on_tls_init(connection_hdl conn)
     {
-        context_ptr ctx = context_ptr(new  asio::ssl::context(asio::ssl::context::tlsv1));
+        context_ptr ctx = context_ptr(new  asio::ssl::context(asio::ssl::context::tlsv12));
```

Comment:
Isn't this forcing TLSv1.2 even if the server supports TLS v1.3?

Usually client libraries are permissive and connect using the most secure connection the server will allow, but there's a good argument for blacklisting ciphers and handshakes that are no longer considered secure to prevent downgrade attacks. Is it possible to choose "any TLS other than v1.0 or v1.1"? If not, I think just permitting all TLS handshakes supported by the server is probably better than hardcoding to v1.2 (eg: `asio::ssl::context::tls`).